marklogic-data-hub/src/main/java/com/marklogic/hub/hubcentral/migration/HubCentralMigrator.java

```diff
@@ -0,0 +1,126 @@
+            return;
+        }
+
+        logger.warn("Beginning migration of entity models in entities directory");
+
+        ObjectWriter writer = mapper.writerWithDefaultPrettyPrinter();
+        boolean atLeastOneEntityModelWasMigrated = false;
+
+        for (JsonNode entityModel : entityManager.getEntitiesJsonNode()) {
+            ObjectNode entityModelNode = (ObjectNode) entityModel;
+            if(entityModelRequiresMigration(entityModelNode)) {
+                atLeastOneEntityModelWasMigrated = true;
+                String title = entityModelNode.get("info").get("title").asText();
+
+                ObjectNode entityTypeNode = (ObjectNode) entityModelNode.get("definitions").get(title);
+                ObjectNode entityTypePropertiesNode = (ObjectNode) entityTypeNode.get("properties");
+
+                List<String> elementRangeIndex = mapper.convertValue(entityTypeNode.get("elementRangeIndex"), ArrayList.class);
+                List<String> rangeIndex = mapper.convertValue(entityTypeNode.get("rangeIndex"), ArrayList.class);
+                elementRangeIndex.removeAll(rangeIndex);
+                rangeIndex.addAll(elementRangeIndex);
+
+                entityTypePropertiesNode.fieldNames().forEachRemaining(propertyName -> {
+                    if(rangeIndex.contains(propertyName)) {
+                        ObjectNode entityPropertyNode = (ObjectNode) entityTypePropertiesNode.get(propertyName);
+                        if(!isStructuredTypeProperty(entityPropertyNode)) {
+                            entityPropertyNode.put("facetable", true);
+                        }
+                    }
+                });
+                entityTypeNode.remove(Arrays.asList("elementRangeIndex", "rangeIndex"));
+
+                File entityModelFile = Paths.get(hubProject.getHubEntitiesDir().toString(), title + EntityManager.ENTITY_FILE_EXTENSION).toFile();
+                try {
+                    writer.writeValue(entityModelFile, entityModelNode);
+                    logger.warn(format("Entity Model '%s' was successfully migrated", entityModelFile));
+                } catch (IOException e) {
+                    logger.error(format("Entity Model '%s' migration failed; cause: %s", entityModelFile, e.getMessage()), e);
+                }
+            }
+        }
+
+        if (atLeastOneEntityModelWasMigrated) {
+            logger.warn("Finished migrating entity models.");
+            logger.warn("Please examine the migrated entities and check the entity type properties defined as element range indexes and path range indexes are having facetable set to true");
+        } else {
+            logger.warn("No entity models required migration, so no project files were modified");
+        }
+    }
+
+    protected boolean entityModelRequiresMigration(ObjectNode entityModelNode) {
+        String firstLevelEntityTypeName= entityModelNode.get("info").get("title").asText();
```

Comment:
Same thing here - be very pessimistic about things not existing. Also check for pathRangeIndex.